contracts/src/periphery/cert/router/EigenDACertVerifierRouter.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import {IEigenDACertVerifier} from "src/periphery/cert/interfaces/IEigenDACertVerifier.sol";
+import {IEigenDACertVerifierRouter} from "src/periphery/cert/interfaces/IEigenDACertVerifierRouter.sol";
+import {OwnableUpgradeable} from "lib/openzeppelin-contracts-upgradeable/contracts/access/OwnableUpgradeable.sol";
+
+contract EigenDACertVerifierRouter is IEigenDACertVerifierRouter, OwnableUpgradeable {
+    mapping(uint32 => address) public certVerifiers;
+    uint32[] public certVerifierRBNs;
+
+    event CertVerifierAdded(uint32 indexed rbn, address indexed certVerifier);
+
+    error RBNNotInFuture(uint32 rbn);
+    error RBNNotGreaterThanLast(uint32 rbn);
+    error InvalidCertLength();
+    error NoCertVerifierAvailable();
+    error NoCertVerifierFound(uint32 rbn);
+
+    /// IEigenDACertVerifierRouter ///
+
+    function checkDACert(bytes calldata certBytes) external view returns (uint8) {
+        return IEigenDACertVerifier(getCertVerifierAt(_getRBN(certBytes))).checkDACert(certBytes);
+    }
+
+    function getCertVerifierAt(uint32 rbn) public view returns (address) {
+        return certVerifiers[_findClosestRegisteredRBN(rbn)];
+    }
+
+    /// ADMIN ///
+
+    function initialize(address _initialOwner) external initializer {
+        _transferOwnership(_initialOwner);
+    }
+
+    function addCertVerifier(uint32 rbn, address certVerifier) external onlyOwner {
+        if (rbn <= block.number) {
+            revert RBNNotInFuture(rbn);
+        }
+        if (certVerifierRBNs.length > 0 && rbn <= certVerifierRBNs[certVerifierRBNs.length - 1]) {
+            revert RBNNotGreaterThanLast(rbn);
+        }
+        certVerifiers[rbn] = certVerifier;
+        certVerifierRBNs.push(rbn);
+        emit CertVerifierAdded(rbn, certVerifier);
+    }
+
+    /// INTERNAL ///
+
+    function _getRBN(bytes calldata certBytes) internal pure returns (uint32) {
+        if (certBytes.length < 36) {
+            revert InvalidCertLength();
+        }
+        return abi.decode(certBytes[32:36], (uint32));
```

Comment:
Does the use of packed ABI encoding vs regular ABI encoding affect the positional assumption?

Reply:
I was wrong. It is positions 32:64 because of padding. I added a test, so this shouldn't be an issue anymore

Reply:
This was indeed wrong. https://github.com/Layr-Labs/eigenda/pull/1518/commits/887ffae4cb5adf2da3f9aaa4062803dbf9752263